server/netty-server/loom/src/main/scala/sttp/tapir/server/netty/loom/NettySyncServerInterpreter.scala

```diff
@@ -1,29 +1,44 @@
 package sttp.tapir.server.netty.loom
 
+import internal.*
+import _root_.ox.Ox
+import sttp.capabilities.WebSockets
+import sttp.monad.syntax._
 import sttp.tapir.server.ServerEndpoint
-import sttp.tapir.server.netty.internal.{NettyToResponseBody, NettyServerInterpreter, RunAsync}
+import sttp.tapir.server.interceptor.reject.RejectInterceptor
+import sttp.tapir.server.interceptor.RequestResult
+import sttp.tapir.server.interpreter.{BodyListener, FilterServerEndpoints, ServerInterpreter}
+import sttp.tapir.server.netty.internal.{NettyBodyListener, RunAsync}
+import sttp.tapir.server.netty.{NettyResponse, NettyServerRequest, Route}
 
-trait NettySyncServerInterpreter {
+trait NettySyncServerInterpreter:
   def nettyServerOptions: NettySyncServerOptions
 
+  /** Requires implicit supervision scope (Ox), because it needs to know in which scope it can start background forks in the Web Sockets
+    * processor.
+    */
   def toRoute(
-      ses: List[ServerEndpoint[Any, Id]]
-  ): IdRoute = {
-    NettyServerInterpreter.toRoute[Id](
-      ses,
-      nettyServerOptions.interceptors,
+      ses: List[ServerEndpoint[OxStreams & WebSockets, Id]]
+  )(using Ox): IdRoute =
```

Comment:
alternatively, we can simply pass the OxDispatcher instance to where it's needed, further narrowing down the places where threads can be started?